### PR TITLE
PS-5964: Introduce Azure Pipelines (8.0)

### PR DESCRIPTION
**Pull request by inikep**
Add azure-pipelines.yml

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+      echo --- Update list of packages and download dependencies
+      sudo -E apt-add-repository -y "ppa:ubuntu-toolchain-r/test"
+      sudo -E apt-get -yq update >> ~/apt-get-update.log 2>&1
+
+      sudo -E apt-get -yq --no-install-suggests --no-install-recommends --allow-unauthenticated install $CXX $PACKAGES cmake cmake-curses-gui ccache bison libncurses5-dev libaio-dev libmecab-dev libnuma-dev liblzma-dev libssl-dev libreadline-dev libpam-dev || exit 1;
+      if [[ "$INVERTED" != "ON" ]]; then
+        sudo -E apt-get -yq --no-install-suggests --no-install-recommends --allow-unauthenticated install libevent-dev libeditline-dev liblz4-dev libre2-dev protobuf-compiler libprotobuf-dev libprotoc-dev libicu-dev || exit 1;
+      fi;
+      sudo ln -s $(which ccache) /usr/lib/ccache/$CC;
+      sudo ln -s $(which ccache) /usr/lib/ccache/$CXX;
```
Comment (percona-ysorokin) [inline azure-pipelines.yml]: BTW, this is how I did this in fb-mysql-8.0.17
https://github.com/percona-ysorokin/mysql-5.6/blob/f8e03f99459827437f118110192735fd22b51c4e/azure-pipelines.yml#L121
No need to create ccache links there.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+  - script: |
+      echo --- Set cmake parameters
+      CMAKE_OPT="
+        -DCMAKE_BUILD_TYPE=$(BuildType)
+        -DBUILD_CONFIG=mysql_release
+        -DFEATURE_SET=community
+        -DENABLE_DOWNLOADS=1
+        -DDOWNLOAD_BOOST=1
+        -DWITH_BOOST=../deps
+        -DWITH_KEYRING_VAULT=OFF
```
Comment (percona-ysorokin) [inline azure-pipelines.yml]: Why `OFF`?

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+        -DWITH_PAM=ON
+        -DWITH_TOKUDB=ON
+        -DWITH_ROCKSDB=ON
+        -DMYSQL_MAINTAINER_MODE=OFF
```
Comment (percona-ysorokin) [inline azure-pipelines.yml]: Why `OFF`?

File: azure-pipelines.yml [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+          -DWITH_EXAMPLE_STORAGE_ENGINE=ON
+          -DWITH_FEDERATED_STORAGE_ENGINE=OFF
+          -DWITHOUT_PERFSCHEMA_STORAGE_ENGINE=ON
+          -DWITH_INNODB_MEMCACHED=ON
+        "
+      fi
+
+      echo --- CMAKE_OPT=\"$CMAKE_OPT\"
+      mkdir bin; cd bin
+      CC=$CC CXX=$CXX cmake .. $CMAKE_OPT || exit 1
```
Comment (percona-ysorokin) [inline azure-pipelines.yml:233]: For some reason, in my experiments `cmake` never returned error return code when finished with an error :(

File: azure-pipelines.yml (hunk 1 above)
Comment (inikep) [reply azure-pipelines.yml]: `CMAKE_*_COMPILER_LAUNCHER` requires CMake 3.4 and it was a problem for `Trusty` but now it should be fine.

File: azure-pipelines.yml (hunk 3 above)
Comment (inikep) [reply azure-pipelines.yml]: updated

File: azure-pipelines.yml (hunk 2 above)
Comment (inikep) [reply azure-pipelines.yml]: updated

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Pull request merged**